server/lib/util.js

```diff
@@ -182,3 +184,27 @@ function convertStringToNumber(obj) {
   });
   return obj;
 }
+
+/*
+ * rename an object's keys
+ */
+
+function renameKeys(objs, newKeys) {
```

Comment:
I think you can use [lodash](https://lodash.com/docs/4.17.10) quite effectively here.  I can rewrite this to be two lines:

```js
const keyMap = _.isString(newKeys) ? JSON.parse(newKeys) : newKeys;
return objs.map(row => _.mapKeys(row, (value, key) => keyMap[key] || key));
```

It works like this:
 1. Parses the JSON if necessary to get the `{ column_id : translatedKey }` map.
 2. Loops through the database rows with `objs.map()`.
 3. For each row object, it uses [lodash's mapKeys](https://lodash.com/docs/4.17.10#mapKeys) function to loop through the keys and replaces the key with a new value.  The new value will either be the translated key found in `keyMap[key]` or itself.